server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala

```diff
@@ -42,54 +43,78 @@ class AddressController @Inject()(esRepo : ElasticsearchRepository)(implicit ec
+    val tokens = AddressTokens(
+      uprn = "",
+      buildingNumber = input.substring(0, 2),
+      postcode = regex.findFirstIn(input).getOrElse("Not recognised")
+    )
+    println(tokens)
 
-    logger info s"#addressQuery parsed: postcode: $postcode , buildingNumber: $buildingNumber"
+    logger info s"#addressQuery parsed: postcode: ${tokens.postcode} , buildingNumber: ${tokens.buildingNumber}"
 
-    Future fromTry format.toAddressScheme flatMap {
-      case PostcodeAddressFile(str) => esRepo.queryAddress(buildingNumber, postcode) map addressResponse
+    Future.fromTry(format.toAddressScheme()).flatMap {
+      case PostcodeAddressFile(str) => esRepo.queryAddress(tokens).map {
+        addresses => addressResponse(tokens, addresses)
+      }
       case _ => asyncWrongFormatResponse
     }
   }
 
+  private def addressResponse(tokens: AddressTokens, addresses: Seq[PostcodeAddressFileAddress]): Result = {
+    val response = AddressBySearchResponseContainer(
+      AddressResponse(
+        tokens,
+        addresses = addresses.map(AddressResponseAddress.fromPafAddress),
+        limit = 10,
+        offset = 0,
+        total = addresses.size
+      ),
+      AddressResponseStatus(
+        code = 200,
+        message = "Ok"
+      ),
+      errors = Seq()
+    )
+    Ok(Json.toJson(response))
+  }
+
   /**
     * UPRN query API
     *
     * @param uprn
     * @param format
     * @return
     */
-  def uprnQuery(
-    uprn : String,
-    format : String
-  ) : Action[AnyContent] = Action async { implicit req =>
-    logger info s"#uprnQuery called with uprn: $uprn , format: $format"
-    Future fromTry format.toAddressScheme flatMap {
-      case PostcodeAddressFile(str) => esRepo queryUprn uprn map addressResponse
+  def uprnQuery(uprn: String, format: String): Action[AnyContent] = Action async { implicit req =>
+    logger info s"#uprnQuery request called with uprn: $uprn , format: $format"
+    Future.fromTry(format.toAddressScheme()).flatMap {
+      case PostcodeAddressFile(str) => esRepo.queryUprn(uprn).map {
+        case Some(address) => Ok(Json.toJson(
+          AddressByUprnResponseContainer(
+           address = AddressResponseAddress.fromPafAddress(address),
+            AddressResponseStatus(
+              code = 200,
+              message = "Ok"
+            ),
+            errors = Seq()
+          )
+        ))
+        case None => NotFound("Not found")
+      }
       case _ => asyncWrongFormatResponse
     }
   }
 
-  private def addressResponse(addresses : Seq[PostcodeAddressFileAddress]) : Result = {
-    logger info "#addressQuery got a response from es"
-    Ok(
-      Json toJson(
-        addresses map PostcodeAddressFileReplyUnit.fromPostcodeAddressFileAddress
-      )
-    )
-  }
 
-  private val asyncWrongFormatResponse : Future[Result] = Future successful Ok("Wrong format")
+  private val asyncWrongFormatResponse: Future[Result] = Future.successful(Ok("Wrong format"))
```

Comment:
I think this is even better to be changed to AddressByUprnResponseFormat with Option for the address and an error in the errors array.